src/runtime/button-api.js

```diff
@@ -152,21 +169,26 @@ export class ButtonApi {
   }
 
   /**
-   * @param {!./deps.DepsDef} deps
    * @param {!Element} button
    * @param {!../api/subscriptions.ButtonOptions|function()} optionsOrCallback
    * @param {function()=} opt_callback
    * @return {!Element}
    */
-  attachSmartButton(deps, button, optionsOrCallback, opt_callback) {
+  attachSmartButton(button, optionsOrCallback, opt_callback) {
     const options = this.getOptions_(optionsOrCallback);
     const callback = /** @type {function()} */
         (this.getCallback_(optionsOrCallback, opt_callback));
 
     // Add required CSS class, if missing.
     button.classList.add('swg-smart-button');
 
+    let analyticsRequest = null;
+    if (this.deps_.config().analyticsMode == AnalyticsMode.IMPRESSIONS) {
+      // isFromUserAction = False
+      analyticsRequest = this.analyticsService_.createLogRequest(
```

Comment:
I'd remove this from here. Since SmartSubscriptionButtonApi has deps, it can get access to AnalyticsService directly.

Reply:
Done.